src/cli-validator/utils/fixProblems.js

```diff
@@ -0,0 +1,229 @@
+    ) {
+      paramsToPrepend[pathstr].splice(
+        paramsToPrepend[pathstr].findIndex(compFunc),
+        1
+      );
+    }
+  };
+
+  const getObjectFromPath = path => {
+    const pathcopy = path.slice(0);
+    let obj = originalJSON;
+    let parent, pParent, ppParent;
+    while (pathcopy.length) {
+      ppParent = pParent;
+      pParent = parent;
+      parent = obj;
+      obj = obj[pathcopy.shift()];
+    }
+    return { obj, parent, pParent, ppParent };
+  };
+
+  // For each problem type and for every problem
+  types.forEach(type => {
+    each(results[type], problems => {
+      problems.forEach(problem => {
+        const message = problem.message.split(':')[0];
+        let path = problem.path;
+        // path needs to be an array to get the line number
+        if (!Array.isArray(path)) {
+          path = path.split('.');
+        }
+        if (path[path.length - 1].includes('[')) {
+          path.push(path[path.length - 1].split(']')[0].split('[')[1]);
+          path[path.length - 2] = path[path.length - 2].split(['['])[0];
+        }
+        const { obj, parent, ppParent } = getObjectFromPath(path);
+        if (message.includes('Schema must have a non-empty description')) {
+          // obj['description'] = path[path.length - 1];
+        } else if (
+          message.includes(
+            'Schema properties must have a description with content in it'
+          )
+        ) {
+          // parent['description'] = path[path.length - 2];
+        } else if (
+          message.includes('Definition was declared but never used in document')
+        ) {
+          definitionsToDelete[path.join(',')] = true;
+        } else if (
+          message.includes(
+            'Common path parameters should be defined on path object'
+          )
+        ) {
+          const paramToAdd = parent[path[path.length - 1]];
+          if ('parameters' in ppParent) {
+            const exists = ppParent['parameters'].some(
+              param => param.name === paramToAdd.name
+            );
+            if (!exists) {
+              ppParent['parameters'].push(paramToAdd);
+            }
+          } else {
+            ppParent['parameters'] = [paramToAdd];
+          }
+          markForDeletion(path);
+          unmarkIfMarkedForMoving(path, parent);
+        } else if (
+          message.includes(
+            'Rely on the `securitySchemas` and `security` fields to specify authorization methods.'
+          )
+        ) {
+          if (!('security' in originalJSON)) {
+            originalJSON['security'] = [{ IAM: [] }];
+            originalJSON['components']['securitySchemes'] = {
+              IAM: {
+                type: 'apiKey',
+                name: 'Authorization',
+                description: obj['description']
+                  ? obj['description']
+                  : 'Your IBM Cloud IAM access token.',
+                in: 'header'
+              }
```

Comment:
This fix is very specific to IBM Cloud services.  I think some additional logic is needed here.